src/main/java/com/coreos/jetcd/internal/impl/ClientConnectionManager.java

```diff
@@ -126,6 +127,17 @@ ExecutorService getExecutorService() {
     return configureStub(supplier.apply(getChannel()), getToken());
   }
 
+  /**
+   * create and add a new token to channel's head.
+   *
+   * @param supplier the stub supplier
+   * @param <T> the type of stub
+   * @return the attached stub
+   */
+  <T extends AbstractStub<T>> T newStubWithNewToken(Function<ManagedChannel, T> supplier) {
+    return configureStub(supplier.apply(getChannel()), generateToken(getChannel()));
```

Comment:
yep, the shared one expired. I think we need to think more about how shared token should be used and updated. We could update the shared token with the newly generated token.